### PR TITLE
Azure: Add Availability Zones

### PR DESCRIPTION
**Pull request by sylr**
##### SUMMARY
Implement Availability Zones for virtual machines and virtual machine scale sets

##### ISSUE TYPE
Feature Pull Request

##### COMPONENT NAME
azure

### CODE REVIEW

**Comment by Fred-sun** [comment]
@sylr Thanks for your contribution。 When you finished changed. I will try the feature and push maintainer to review? Thanks!

**Comment by yungezz** [comment]
@zikalino  pls review the pr as you want to.

**Comment by Fred-sun** [comment]
@zikalino

**Comment by Fred-sun** [comment]
@zikalino  Could you help review this PR and push for merge ? Another PR duplicate of this, if we merged this, we can close  relate issue and PR? Thanks!

**Comment by Fred-sun** [comment]
@sylr Thanks for your contribution, Could you help re-check this PR and solve the conflicting files? if you finished, I will push this for review. Thanks!

**Comment by sylr** [comment]
@Fred-sun rebased.

Please review/merge this before new conflicts happen.

**Comment by Fred-sun** [comment]
@yuwzho @zikalino @yungezz Please help review this PR and push this for merged! Thanks!

File: lib/ansible/modules/cloud/azure/azure_rm_virtualmachine.py
Comment (zikalino) [inline lib/ansible/modules/cloud/azure/azure_rm_virtualmachine.py:1044]: I think we should have warning instead of fail here, but i think it can be done later. no need to hold this pr.

**Review by zikalino — APPROVED** [review]

**Comment by zikalino** [comment]
shipit

**Comment by yungezz** [comment]
rebuild_merge

**Comment by nomadcz** [comment]
Hi. Is this feature available also for managed disks (azure_rm_managed_disk) ? (similar to az disk create --name xxx --zone X ...) ?

**Comment by yungezz** [comment]
@nomadcz not yet. could you pls create an issue so that we can track the request? thanks.

**Pull request merged**
